homeassistant/components/hunterdouglas_powerview/button.py

```diff
@@ -0,0 +1,142 @@
+"""Buttons for Hunter Douglas Powerview advanced features."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SHADE_DATA,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PowerviewButtonDescriptionMixin:
+    """Mixin to describe a Button entity."""
+
+    press_action: Callable
+
+
+@dataclass
+class PowerviewButtonDescription(
+    ButtonEntityDescription, PowerviewButtonDescriptionMixin
+):
+    """Class to describe a Button entity."""
+
+
+BUTTONS: Final = [
+    PowerviewButtonDescription(
+        key="calibrate",
+        name="Calibrate",
+        icon="mdi:swap-vertical-circle-outline",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: shade.calibrate(),
+    ),
+    PowerviewButtonDescription(
+        key="identify",
+        name="Identify",
+        icon="mdi:crosshairs-question",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: shade.jog(),
+    ),
+    PowerviewButtonDescription(
+        key="update",
+        name="Force Update",
+        icon="mdi:autorenew",
+        device_class=ButtonDeviceClass.UPDATE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        press_action=lambda shade, button: button.async_force_refresh(),
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set buttons for device."""
+
+    pv_data = hass.data[DOMAIN][entry.entry_id]
+    room_data: dict[str | int, Any] = pv_data[PV_ROOM_DATA]
+    shade_data = pv_data[PV_SHADE_DATA]
+    pv_request = pv_data[PV_API]
+    coordinator: PowerviewShadeUpdateCoordinator = pv_data[COORDINATOR]
+    device_info: dict[str, Any] = pv_data[DEVICE_INFO]
```

Comment:
Not related to this PR but for the future, `pv_data` should get converted to a dataclass instead of a `dict`.

This integration is quite old so dataclasses were not a thing when it was created so there is a lot of legacy here.